interop/executor.go

```diff
@@ -135,8 +135,21 @@ func (e *Executor) verifySignature(stx tx.SignedTx) error {
 	if err != nil {
 		return errors.New("failed to get admin from L1")
 	}
-	if sequencer != signer {
-		return errors.New("unexpected signer")
+
+	// Attempt to retrieve the authorized proof signer for the given rollup, if one exists
+	authorizedProofSigner, hasKey := e.config.ProofSigners[stx.Tx.RollupID]
+
+	// If an authorized proof signer is defined and matches the signer, no further checks are needed
+	if hasKey {
+		// If an authorized proof signer exists but does not match the signer, return an error.
+		if authorizedProofSigner != signer {
+			return fmt.Errorf("the signer %s did not match the authorized signer %s", signer, authorizedProofSigner)
```

Comment:
Nitpick, just to follow the same standard as the old error:

```suggestion
			return fmt.Errorf("unexpected signer: expected authorized signer %s, but got %s", authorizedProofSigner, signer)
```